rest/route/distro.go

```diff
@@ -240,3 +289,41 @@ func (dgh *distroGetHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return resp
 }
+
+////////////////////////////////////////////////////////////////////////
+
+func validateToService(ctx context.Context, apiDistro *model.APIDistro, resourceId string, isNewDistro bool) (*distro.Distro, gimlet.Responder) {
+	i, err := apiDistro.ToService()
+	if err != nil {
+		return nil, gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "API Error converting from model.APIDistro to distro.Distro"))
+	}
+	d := i.(*distro.Distro)
+
+	id := model.FromAPIString(apiDistro.Name)
+	if resourceId != id {
+		return nil, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("Distro name is immutable; cannot rename distro resource '%s'", resourceId),
+		})
+	}
+
+	vErrors, err := validator.CheckDistro(ctx, d, &evergreen.Settings{}, isNewDistro)
+	if err != nil {
+		return nil, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    err.Error(),
+		})
+	}
+	if len(vErrors) != 0 {
+		errors := []string{}
+		for _, v := range vErrors {
+			errors = append(errors, v.Message)
+		}
+		return nil, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    strings.Join(errors, ", "),
+		})
+	}
```

Comment:
vErrors has a string method that you can use.

Reply:
Changed to => Message: vErrors.String()